From a4b6e2bc62dce3d84db4e5aed4b914a0b4b5a15d Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Mon, 3 Jul 2017 16:49:51 +0200
Subject: [PATCH] Respect already defined by build environment CFLAGS. Added
 base LDFLAGS (will be overriden via build environment). Fixed incorrect chmod
 on shared library and icons. Preserve timestamps on installed files.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 Makefile.common | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile.common b/Makefile.common
index c800bd1..46692c0 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -2,9 +2,10 @@
 
 COMPILE.c = $(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
 
-CFLAGS+=-Wall -g -O0 -Werror
+CFLAGS ?= -Wall -g -O0 -Werror
 CFLAGS += -DPURPLE_PLUGINS 
 CFLAGS += -DGLIB_DISABLE_DEPRECATION_WARNINGS
+LDFLAGS ?= -Wl,-z,relro
 
 # generate .d files when compiling
 CPPFLAGS += -MMD
@@ -28,14 +29,14 @@ clean:
 
 install:
 	mkdir -p $(DESTDIR)$(PLUGIN_DIR_PURPLE)
-	install -m 664 $(TARGET) $(DESTDIR)$(PLUGIN_DIR_PURPLE)
+	install -m 0755 -p $(TARGET) $(DESTDIR)$(PLUGIN_DIR_PURPLE)
 	for i in 16 22 48; do \
 	    mkdir -p $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/$$i; \
-	    install -m 664 matrix-$${i}px.png $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/$$i/matrix.png; \
+	    install -m 0644 -p matrix-$${i}px.png $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/$$i/matrix.png; \
 	done
 
 $(TARGET): $(OBJECTS)
-	$(LINK.o) -shared $^ $(LOADLIBES) $(LDLIBS) -o $@
+	$(LINK.o) -shared $^ $(LOADLIBES) $(CFLAGS) $(LDFLAGS) $(LDLIBS) -o $@
 
 %.o: %.c
 	$(COMPILE.c) $(OUTPUT_OPTION) $<

-- 
2.9.4

